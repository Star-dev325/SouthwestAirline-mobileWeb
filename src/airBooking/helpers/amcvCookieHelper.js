// @flow
import _ from 'lodash';
import * as Cookie from 'src/shared/swa-persistence/cookie';

import { buildPathWithQuery } from 'src/shared/helpers/pathUtils';

const AMCV_COOKIE_NAME = 'AMCV_65D316D751E563EC0A490D4C%40AdobeOrg';
const AMCV_COOKIE_KEY_SEPARATOR = 'MID';
const AMCV_COOKIE_PIPE_SEPARATOR = '|';
const CHASE_URL_KEYWORD = 'creditcard';

export type ChaseAppendParams = {
  isChaseCombo?: boolean,
  pageId?: string
};

export const getMcvid = () => {
  const value = Cookie.getValue(AMCV_COOKIE_NAME);

  return _.chain(value).split(AMCV_COOKIE_KEY_SEPARATOR).nth(1).split(AMCV_COOKIE_PIPE_SEPARATOR).nth(1).value() || '';
};

// Deprecated - please use appendParamsIfChaseUrl instead
export const addMvcidToChaseUrl = (url: string, isChase: boolean) => {
  let updateUrl = url;

  if (!_.isEmpty(url) && (isChase || url.includes(CHASE_URL_KEYWORD))) {
    const mcvid = getMcvid();

    if (!_.isEmpty(mcvid)) {
      updateUrl = `${url}&mcvid=${mcvid}`;
    }
  }

  return updateUrl;
};

export const appendParamsIfChaseUrl = (target: string = '', params: ChaseAppendParams = {}) => {
  const { isChaseCombo = false, pageId = '' } = params;
  const query = {};
  const isChaseUrl = _.includes(target, CHASE_URL_KEYWORD) || isChaseCombo;
  const mcvid = isChaseUrl && getMcvid();

  mcvid && _.set(query, 'mcvid', mcvid);
  pageId && _.set(query, 'pageId', pageId);

  return buildPathWithQuery(target, query);
};
