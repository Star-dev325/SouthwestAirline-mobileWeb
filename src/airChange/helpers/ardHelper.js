import _ from 'lodash';
import { AIR_CHANGE_ARD_CODE } from 'src/airChange/constants/airChangeConstants';
import ArdErrorMessages from 'src/airChange/constants/ardErrorMessages';

const { TITLE, MESSAGE } = ArdErrorMessages;

export default {
  isARD(errorResponse) {
    return _.get(errorResponse, 'responseJSON.code') === AIR_CHANGE_ARD_CODE;
  },
  transformARDError(errorResponse) {
    return _.merge(errorResponse, {
      responseJSON: {
        title: TITLE,
        message: MESSAGE
      }
    });
  }
};
