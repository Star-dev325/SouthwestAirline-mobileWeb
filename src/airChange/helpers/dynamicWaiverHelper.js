// @flow

import dayjs from 'dayjs';
import _ from 'lodash';
import type {
  DynamicWaiverType,
  FormDataType,
  SearchRequest,
  SelectedBounds
} from 'src/airChange/flow-typed/airChange.types';

export const isDynamicWaiverEligible = (
  dynamicWaivers: Array<DynamicWaiverType>,
  searchRequest: SearchRequest | FormDataType,
  selectedBounds: SelectedBounds
) =>
  isWithinDynamicWaiverAlternativeCities(dynamicWaivers, searchRequest, selectedBounds) &&
  isWithinDynamicWaiverDateRange(dynamicWaivers, searchRequest, selectedBounds);

export const isWithinDynamicWaiverAlternativeCities = (
  dynamicWaivers: Array<DynamicWaiverType>,
  searchRequest: SearchRequest | FormDataType,
  selectedBounds: SelectedBounds
) => {
  const isRoundTripChange = _.size(_.filter(selectedBounds)) === 2;

  return getSelectedDynamicWaiverBounds(dynamicWaivers, selectedBounds).every(({ dynamicWaiver, isFirstBound }) => {
    const { alternativeDepartureCities, alternativeArrivalCities } = dynamicWaiver;

    const fromStation = isRoundTripChange && !isFirstBound ? searchRequest.to : searchRequest.from;
    const isFromStationValid =
      _.isEmpty(alternativeDepartureCities) || _.includes(alternativeDepartureCities, fromStation);

    const toStation = isRoundTripChange && !isFirstBound ? searchRequest.from : searchRequest.to;
    const isToStationValid = _.isEmpty(alternativeArrivalCities) || _.includes(alternativeArrivalCities, toStation);

    return isFromStationValid && isToStationValid;
  });
};

export const isWithinDynamicWaiverDateRange = (
  dynamicWaivers: Array<DynamicWaiverType>,
  searchRequest: SearchRequest | FormDataType,
  selectedBounds: SelectedBounds
) =>
  getSelectedDynamicWaiverBounds(dynamicWaivers, selectedBounds).every(({ dynamicWaiver, isFirstBound }) => {
    const { calculatedStartDate, calculatedEndDate } = dynamicWaiver;

    const date = isFirstBound
      ? _.get(searchRequest, 'departureAndReturnDate.departureDate')
      : _.get(searchRequest, 'departureAndReturnDate.returnDate');

    return dayjs(date).isBetween(calculatedStartDate, calculatedEndDate, 'day', '[]');
  });

const getSelectedDynamicWaiverBounds = (dynamicWaivers: Array<DynamicWaiverType>, selectedBounds: SelectedBounds) =>
  _.chain(selectedBounds)
    .omitBy((value) => !value)
    .map((value, key) =>
      (key === 'firstbound'
        ? {
          dynamicWaiver: _.first(dynamicWaivers),
          isFirstBound: true
        }
        : {
          dynamicWaiver: _.get(dynamicWaivers, '1'),
          isFirstBound: false
        })
    )
    .filter((bound) => !_.isEmpty(bound.dynamicWaiver))
    .value();
