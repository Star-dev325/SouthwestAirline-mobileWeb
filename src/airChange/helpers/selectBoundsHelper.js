import _ from 'lodash';

const SelectBoundsHelper = {
  hasBothBoundsSelected(selectBounds) {
    return _.isEqual(selectBounds, ['outbound', 'inbound']);
  },
  hasOutboundSelectedOnly(selectBounds) {
    return _.isEqual(selectBounds, ['outbound']);
  },
  hasInboundSelectedOnly(selectBounds) {
    return _.isEqual(selectBounds, ['inbound']);
  }
};

export default SelectBoundsHelper;
