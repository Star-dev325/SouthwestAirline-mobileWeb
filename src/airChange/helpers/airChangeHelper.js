// @flow
import { toNumberFromFormattedString } from 'src/shared/helpers/currencyValueHelper';

import type { BoundSelection } from 'src/shared/flow-typed/boundSelect.types';

export const hasEnoughPointsForFare = (amount: string, accountRedeemablePoints: number) => {
  const totalPointsCost = toNumberFromFormattedString(amount);

  return accountRedeemablePoints >= totalPointsCost;
};

export const getIsReaccomCoTerminalEligible = (reaccomBoundSelections: Array<BoundSelection>) =>
  !!reaccomBoundSelections.find(
    (bound) =>
      bound.alternateReaccomOriginationAirportCodes ||
      bound.alternateReaccomDestinationAirportCodes ||
      bound.shoppingDates ||
      bound.multiSelectShoppingDates
  );
