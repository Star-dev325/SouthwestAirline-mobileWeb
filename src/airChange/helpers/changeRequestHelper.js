// @flow

import _ from 'lodash';

import { removeSeparator } from 'src/shared/helpers/separatorHelper';
import { generateNewCreditCardInfo, generateUatpCardInfo } from 'src/shared/helpers/creditCardHelper';
import { isCurrencyAmountZero } from 'src/shared/helpers/travelFundsHelper';
import { APPLE_PAY } from 'src/shared/constants/creditCardTypes';

import { NEW_CREDIT_CARD_ID, PAY_PAL_CARD_ID, APPLE_PAY_CARD_ID } from 'src/shared/constants/creditCardConstants';

import type { FlightChangeRequestDataType } from 'src/airChange/flow-typed/airChange.types';

export const generateChangeRequest = (
  flightChangeRequestData: FlightChangeRequestDataType,
  changeConfirmationPageLink: Link
) => {
  const {
    contactMethodInfo: { contactMethod, email, phoneNumber, phoneCountryCode, declineNotifications },
    paymentInfo,
    emailReceiptTo,
    moneyTotalFare,
    refundMethod,
    paymentRequired,
    securityCode,
    fundsAppliedToken,
    applePayCard
  } = flightChangeRequestData;
  const selectedCardId = _.get(paymentInfo, 'selectedCardId');

  const requestBody: { [string]: * } = { emailReceiptTo };
  const needContactMethod = _.toBoolean(declineNotifications);

  if (needContactMethod) {
    requestBody.declineNotifications = true;
  } else {
    requestBody.contactInformation = {
      email,
      phone: {
        countryCode: phoneCountryCode,
        number: removeSeparator(phoneNumber)
      },
      contactMethod
    };
  }

  if (!_.isEmpty(refundMethod)) {
    requestBody.refundMethod = refundMethod;
  }

  if (paymentRequired || fundsAppliedToken) {
    let pointsTotalBaseFare = _.get(flightChangeRequestData, 'flightPricingPage.totals.pointsTotal', null);
    let payment: { [string]: * } = { moneyTotalFare };

    pointsTotalBaseFare = pointsTotalBaseFare
      ? _.pick(pointsTotalBaseFare, ['amount', 'currencyCode'])
      : pointsTotalBaseFare;

    const purchasePaymentInfo = { moneyTotalFare, pointsTotalBaseFare };

    fundsAppliedToken && _.set(payment, 'fundToken', fundsAppliedToken);

    if (!isCurrencyAmountZero(moneyTotalFare)) {
      switch (selectedCardId) {
        case NEW_CREDIT_CARD_ID:
          payment.newCreditCard = generateNewCreditCardInfo(paymentInfo);
          break;
        case PAY_PAL_CARD_ID: {
          const paypalToken = _.get(flightChangeRequestData, 'payPal.token');

          payment.paypal = { paypalToken };
          break;
        }
        case APPLE_PAY_CARD_ID: {
          if (applePayCard && applePayCard.isNativeApplePay) {
            payment = _.merge({}, purchasePaymentInfo, applePayCard.purchaseRequest);
          } else {
            payment.newCreditCard = generateUatpCardInfo(applePayCard, APPLE_PAY.key);
          }
          break;
        }
        default: {
          payment.savedCreditCard = { savedCreditCardId: selectedCardId };
          break;
        }
      }
    }

    !_.isEmpty(securityCode) && _.set(payment, 'savedCreditCard.securityCode', securityCode);
    requestBody.payment = payment;
  }

  return _.merge({}, changeConfirmationPageLink, {
    body: requestBody
  });
};
