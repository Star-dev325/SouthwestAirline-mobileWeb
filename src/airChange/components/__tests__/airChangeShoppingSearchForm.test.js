import { fireEvent, render } from '@testing-library/react';
import dayjs from 'dayjs';
import _ from 'lodash';
import React from 'react';
import { Provider } from 'react-redux';
import AirChangeShoppingSearchForm from 'src/airChange/components/airChangeShoppingSearchForm';
import TripTypes from 'src/shared/constants/tripTypes';
import { getMultiSelectGroup } from 'test/builders/model/multiSelectGroupBuilder';
import { createMockedFormStore } from 'test/unit/helpers/formTestUtils';

let updateSelectedAirportInfoFnMock;

describe('AirChangeShoppingSearchForm', () => {
  const defaultInitialFormData = {
    departureAndReturnDate: {
      departureDate: '2018-04-04',
      returnDate: '2018-04-09'
    },
    from: 'AUS',
    to: 'BOI'
  };

  beforeEach(() => {
    updateSelectedAirportInfoFnMock = jest.fn();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  describe('render', () => {
    it('should render page properly when allowARNKPnrs is true and both bounds are selected', () => {
      const { container } = createComponent({
        allowARNKPnrs: true,
        selectedBounds: { firstbound: true, secondbound: true }
      });

      expect(container).toMatchSnapshot();
    });

    it('should pass originDisabled to origin airport selector', () => {
      const { container } = createComponent();

      expect(container).toMatchSnapshot();
    });

    it('should grey out from and to fields when hasUnaccompaniedMinor is true', () => {
      const { container } = createComponent({ hasUnaccompaniedMinor: true });

      expect(container).toMatchSnapshot();
    });

    it('should grey out from and to fields when trip type is one way and both bounds are enabled', () => {
      const { container } = createComponent({
        searchOptions: {
          departureBoundDisabled: false,
          returnBoundDisabled: false,
          tripType: TripTypes.ONE_WAY.value
        }
      });

      expect(container).toMatchSnapshot();
    });

    it('should grey out from and to fields when trip type is one way and departure bound is disabled', () => {
      const { container } = createComponent({
        searchOptions: {
          departureBoundDisabled: true,
          returnBoundDisabled: false,
          tripType: TripTypes.ONE_WAY.value
        }
      });

      expect(container).toMatchSnapshot();
    });

    describe('when isReaccomCoTerminalEligible is true', () => {
      it('should render correctly', () => {
        const { container } = createComponent({
          isReaccomCoTerminalEligible: true,
          isRoundTrip: true,
          searchOptions: {
            departureBoundDisabled: false,
            returnBoundDisabled: false,
            tripType: TripTypes.ROUND_TRIP.value
          }
        });

        expect(container).toMatchSnapshot();
      });

      it('should render correctly with reaccomCoTerminalDepartureAirports and reaccomCoTerminalReturnAirports', () => {
        const { container } = createComponent({
          isReaccomCoTerminalEligible: true,
          isRoundTrip: true,
          searchOptions: {
            departureBoundDisabled: false,
            reaccomCoTerminalDepartureAirports: getMultiSelectGroup()['New York Area Airports'],
            reaccomCoTerminalReturnAirports: getMultiSelectGroup()['Boston Area Airports'],
            returnBoundDisabled: false,
            tripType: TripTypes.ROUND_TRIP.value
          }
        });

        expect(container).toMatchSnapshot();
      });

      it('should render correctly when isInvalidDepartureDate is true', () => {
        const state = {
          ...defaultInitialFormData,
          departureAndReturnDate: {
            ...defaultInitialFormData.departureAndReturnDate,
            isInvalidDepartureDate: true,
            isInvalidReturnDate: false
          }
        };
        const { container } = createComponent({
          isReaccomCoTerminalEligible: true,
          isRoundTrip: true,
          searchOptions: {
            departureBoundDisabled: false,
            returnBoundDisabled: false,
            tripType: TripTypes.ROUND_TRIP.value
          }
        }, state);

        expect(container).toMatchSnapshot();
      });
    });
  });

  describe('submit', () => {
    it('should not call onSubmit when flight is the same', () => {
      const onSubmitMock = jest.fn();
      const { container } = createComponent({ onSubmit: onSubmitMock });
      const form = container.querySelector('form');

      fireEvent.submit(form);

      expect(onSubmitMock).toBeCalled();
    });
  });

  const createComponent = (props = {}, initialFormData = defaultInitialFormData) => {
    const defaultProps = {
      allAirports: [],
      allowARNKPnrs: false,
      earliestBookableDate: dayjs('2018-04-04'),
      formData: {
        departureAndReturnDate: {
          isInvalidDepartureDate: false,
          isInvalidReturnDate: false
        }
      },
      formId: 'airChangeShoppingSearchForm',
      hasUnaccompaniedMinor: false,
      isReaccomCoTerminalEligible: false,
      isRoundTrip: false,
      lastBookableDate: dayjs('2018-07-09'),
      onSubmit: () => {},
      recentlySearched: [],
      searchOptions: {
        departureBoundDisabled: false,
        reaccomCoTerminalDates: {
          departureEarliestBookableDate: undefined,
          departureLastBookableDate: undefined,
          returnEarliestBookableDate: undefined,
          returnLastBookableDate: undefined
        },
        returnBoundDisabled: false,
        tripType: 'roundTrip'
      },
      selectedBounds: { firstbound: true, secondbound: false },
      updateSelectedAirportInfoFn: updateSelectedAirportInfoFnMock
    };
    const mergedProps = _.merge({}, defaultProps, props);

    return render(
      <Provider store={createMockedFormStore()}>
        <AirChangeShoppingSearchForm initialFormData={initialFormData} {...mergedProps} />
      </Provider>
    );
  };
});
