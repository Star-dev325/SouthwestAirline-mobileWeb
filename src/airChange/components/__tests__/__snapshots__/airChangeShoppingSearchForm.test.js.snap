// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AirChangeShoppingSearchForm render should grey out from and to fields when hasUnaccompaniedMinor is true 1`] = `
<div>
  <form
    class="book-flight-form form widget"
    name="air-change-flight-search"
    novalidate=""
  >
    <fieldset>
      <div
        class="fields grouped"
      >
        <div
          class="field"
        >
          <div
            class="airport-selector"
            data-qa="from"
          >
            <div
              class="clickable-div input"
            >
              <div
                class="input--label"
              >
                <i
                  class="icon icon_airplane-depart"
                />
              </div>
              <div
                class="ellipsis"
              >
                <div
                  class="airport-selector--placeholder-container py3"
                >
                  <div
                    class="airport-selector--placeholder"
                  >
                    From
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="fields grouped"
      >
        <div
          class="field"
        >
          <div
            class="airport-selector"
            data-qa="to"
          >
            <div
              class="clickable-div input"
            >
              <div
                class="input--label"
              >
                <i
                  class="icon icon_airplane-return"
                />
              </div>
              <div
                class="ellipsis"
              >
                <div
                  class="airport-selector--placeholder-container py3"
                >
                  <div
                    class="airport-selector--placeholder"
                  >
                    To
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="fields grouped"
      >
        <div
          class="field"
        >
          <div>
            <div
              class="clickable-div input"
            >
              <div
                class="input--label"
              >
                <i
                  class="icon icon_calender"
                />
              </div>
              <div
                class="ellipsis py3 bold"
                data-qa="depart-and-return-dates"
              >
                4/04/18 - 4/09/18
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="transparent segment"
      >
        <button
          class="button button--fluid larger button--yellow"
          role="submit"
          type="submit"
        >
          AIR_CHANGE__SHOPPING_SEARCH_PAGE__FIND_FLIGHTS_BUTTON
        </button>
      </div>
    </fieldset>
  </form>
</div>
`;

exports[`AirChangeShoppingSearchForm render should grey out from and to fields when trip type is one way and both bounds are enabled 1`] = `
<div>
  <form
    class="book-flight-form form widget"
    name="air-change-flight-search"
    novalidate=""
  >
    <fieldset>
      <div
        class="fields grouped"
      >
        <div
          class="field"
        >
          <div
            class="airport-selector"
            data-qa="from"
          >
            <div
              class="clickable-div input"
            >
              <div
                class="input--label"
              >
                <i
                  class="icon icon_airplane-depart"
                />
              </div>
              <div
                class="ellipsis"
              >
                <div
                  class="airport-selector--placeholder-container py3"
                >
                  <div
                    class="airport-selector--placeholder"
                  >
                    From
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="fields grouped"
      >
        <div
          class="field"
        >
          <div
            class="airport-selector"
            data-qa="to"
          >
            <div
              class="clickable-div input"
            >
              <div
                class="input--label"
              >
                <i
                  class="icon icon_airplane-return"
                />
              </div>
              <div
                class="ellipsis"
              >
                <div
                  class="airport-selector--placeholder-container py3"
                >
                  <div
                    class="airport-selector--placeholder"
                  >
                    To
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="fields grouped"
      >
        <div
          class="field"
        >
          <div>
            <div
              class="clickable-div input"
            >
              <div
                class="input--label"
              >
                <i
                  class="icon icon_calender"
                />
              </div>
              <div
                class="ellipsis py3 bold"
                data-qa="depart-and-return-dates"
              >
                4/04/18 - 4/09/18
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="transparent segment"
      >
        <button
          class="button button--fluid larger button--yellow"
          role="submit"
          type="submit"
        >
          AIR_CHANGE__SHOPPING_SEARCH_PAGE__FIND_FLIGHTS_BUTTON
        </button>
      </div>
    </fieldset>
  </form>
</div>
`;

exports[`AirChangeShoppingSearchForm render should grey out from and to fields when trip type is one way and departure bound is disabled 1`] = `
<div>
  <form
    class="book-flight-form form widget"
    name="air-change-flight-search"
    novalidate=""
  >
    <fieldset>
      <div
        class="fields grouped"
      >
        <div
          class="field"
        >
          <div
            class="airport-selector"
            data-qa="from"
          >
            <div
              class="clickable-div input"
            >
              <div
                class="input--label"
              >
                <i
                  class="icon icon_airplane-depart"
                />
              </div>
              <div
                class="ellipsis"
              >
                <div
                  class="airport-selector--placeholder-container py3"
                >
                  <div
                    class="airport-selector--placeholder"
                  >
                    From
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="fields grouped"
      >
        <div
          class="field"
        >
          <div
            class="airport-selector"
            data-qa="to"
          >
            <div
              class="clickable-div input"
            >
              <div
                class="input--label"
              >
                <i
                  class="icon icon_airplane-return"
                />
              </div>
              <div
                class="ellipsis"
              >
                <div
                  class="airport-selector--placeholder-container py3"
                >
                  <div
                    class="airport-selector--placeholder"
                  >
                    To
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="fields grouped"
      >
        <div
          class="field"
        >
          <div>
            <div
              class="clickable-div input"
            >
              <div
                class="input--label"
              >
                <i
                  class="icon icon_calender"
                />
              </div>
              <div
                class="ellipsis py3 bold"
                data-qa="depart-and-return-dates"
              >
                <div>
                  <span
                    class="gray4"
                    data-qa="disabled-date"
                  >
                    4/04/18
                  </span>
                  <span
                    class="gray4"
                  >
                     - 
                  </span>
                  <span>
                    4/09/18
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="transparent segment"
      >
        <button
          class="button button--fluid larger button--yellow"
          role="submit"
          type="submit"
        >
          AIR_CHANGE__SHOPPING_SEARCH_PAGE__FIND_FLIGHTS_BUTTON
        </button>
      </div>
    </fieldset>
  </form>
</div>
`;

exports[`AirChangeShoppingSearchForm render should pass originDisabled to origin airport selector 1`] = `
<div>
  <form
    class="book-flight-form form widget"
    name="air-change-flight-search"
    novalidate=""
  >
    <fieldset>
      <div
        class="fields grouped"
      >
        <div
          class="field"
        >
          <div
            class="airport-selector"
            data-qa="from"
          >
            <div
              class="clickable-div input"
            >
              <div
                class="input--label"
              >
                <i
                  class="icon icon_airplane-depart"
                />
              </div>
              <div
                class="ellipsis"
              >
                <div
                  class="airport-selector--placeholder-container py3"
                >
                  <div
                    class="airport-selector--placeholder"
                  >
                    From
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="fields grouped"
      >
        <div
          class="field"
        >
          <div
            class="airport-selector"
            data-qa="to"
          >
            <div
              class="clickable-div input"
            >
              <div
                class="input--label"
              >
                <i
                  class="icon icon_airplane-return"
                />
              </div>
              <div
                class="ellipsis"
              >
                <div
                  class="airport-selector--placeholder-container py3"
                >
                  <div
                    class="airport-selector--placeholder"
                  >
                    To
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="fields grouped"
      >
        <div
          class="field"
        >
          <div>
            <div
              class="clickable-div input"
            >
              <div
                class="input--label"
              >
                <i
                  class="icon icon_calender"
                />
              </div>
              <div
                class="ellipsis py3 bold"
                data-qa="depart-and-return-dates"
              >
                4/04/18 - 4/09/18
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="transparent segment"
      >
        <button
          class="button button--fluid larger button--yellow"
          role="submit"
          type="submit"
        >
          AIR_CHANGE__SHOPPING_SEARCH_PAGE__FIND_FLIGHTS_BUTTON
        </button>
      </div>
    </fieldset>
  </form>
</div>
`;

exports[`AirChangeShoppingSearchForm render should render page properly when allowARNKPnrs is true and both bounds are selected 1`] = `
<div>
  <form
    class="book-flight-form form widget"
    name="air-change-flight-search"
    novalidate=""
  >
    <fieldset>
      <div
        class="air-change-shopping-search-arnk-layout--button"
      >
        <button
          class="button button--fluid larger button--yellow"
          role="submit"
          type="submit"
        >
          AIR_CHANGE__SHOPPING_SEARCH_PAGE__FIND_FLIGHTS_BUTTON
        </button>
      </div>
    </fieldset>
  </form>
</div>
`;

exports[`AirChangeShoppingSearchForm render when isReaccomCoTerminalEligible is true should render correctly 1`] = `
<div>
  <form
    class="book-flight-form form widget"
    name="air-change-flight-search"
    novalidate=""
  >
    <fieldset>
      <div
        class="fields grouped"
      >
        <div
          class="field"
        >
          <div
            class="airport-selector"
            data-qa="from"
          >
            <div
              class="clickable-div input"
            >
              <div
                class="input--label"
              >
                <i
                  class="icon icon_airplane-depart"
                />
              </div>
              <div
                class="ellipsis"
              >
                <div
                  class="airport-selector--placeholder-container py3"
                >
                  <div
                    class="airport-selector--placeholder"
                  >
                    From
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="fields grouped"
      >
        <div
          class="field"
        >
          <div
            class="airport-selector"
            data-qa="to"
          >
            <div
              class="clickable-div input"
            >
              <div
                class="input--label"
              >
                <i
                  class="icon icon_airplane-return"
                />
              </div>
              <div
                class="ellipsis"
              >
                <div
                  class="airport-selector--placeholder-container py3"
                >
                  <div
                    class="airport-selector--placeholder"
                  >
                    To
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="fields grouped"
      >
        <div
          class="field"
        >
          <div>
            <div
              class="form-calendar-field py3"
              data-qa="depart-and-return-dates"
            >
              <div
                class="form-calendar-field--departure-date"
              >
                <div
                  class="form-calendar-field--date bold xxlarge"
                >
                  4/04
                </div>
                <div
                  class="form-calendar-field--date-description medium"
                >
                  <span
                    class="italic"
                  >
                    Wed
                  </span>
                  <span>
                    , Apr 04, 2018
                  </span>
                </div>
              </div>
              <i
                class="form-calendar-field--icon icon icon_calender"
              />
              <div
                class="form-calendar-field--return-date"
              >
                <div
                  class="form-calendar-field--date bold xxlarge"
                >
                  4/09
                </div>
                <div
                  class="form-calendar-field--date-description medium"
                >
                  <span
                    class="italic"
                  >
                    Mon
                  </span>
                  <span>
                    , Apr 09, 2018
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="transparent segment"
      >
        <button
          class="button button--fluid larger button--yellow"
          role="submit"
          type="submit"
        >
          AIR_CHANGE__SHOPPING_SEARCH_PAGE__FIND_FLIGHTS_BUTTON
        </button>
      </div>
    </fieldset>
  </form>
</div>
`;

exports[`AirChangeShoppingSearchForm render when isReaccomCoTerminalEligible is true should render correctly when isInvalidDepartureDate is true 1`] = `
<div>
  <form
    class="book-flight-form form widget"
    name="air-change-flight-search"
    novalidate=""
  >
    <fieldset>
      <div
        class="fields grouped"
      >
        <div
          class="field"
        >
          <div
            class="airport-selector"
            data-qa="from"
          >
            <div
              class="clickable-div input"
            >
              <div
                class="input--label"
              >
                <i
                  class="icon icon_airplane-depart"
                />
              </div>
              <div
                class="ellipsis"
              >
                <div
                  class="airport-selector--placeholder-container py3"
                >
                  <div
                    class="airport-selector--placeholder"
                  >
                    From
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="fields grouped"
      >
        <div
          class="field"
        >
          <div
            class="airport-selector"
            data-qa="to"
          >
            <div
              class="clickable-div input"
            >
              <div
                class="input--label"
              >
                <i
                  class="icon icon_airplane-return"
                />
              </div>
              <div
                class="ellipsis"
              >
                <div
                  class="airport-selector--placeholder-container py3"
                >
                  <div
                    class="airport-selector--placeholder"
                  >
                    To
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="fields grouped"
      >
        <div
          class="field"
        >
          <div>
            <div
              class="form-calendar-field py3"
              data-qa="depart-and-return-dates"
            >
              <div
                class="form-calendar-field--departure-date"
              >
                <div
                  class="form-calendar-field--date bold xxlarge"
                >
                  SHARED__CALENDAR__REACCOM_DEPARTURE_DATE_TEXT
                </div>
                <div
                  class="form-calendar-field--description"
                >
                  SHARED__CALENDAR__REACCOM_DATE_DESCRIPTION_TEXT
                </div>
              </div>
              <i
                class="form-calendar-field--icon icon icon_calender"
              />
              <div
                class="form-calendar-field--return-date"
              >
                <div
                  class="form-calendar-field--date bold xxlarge"
                >
                  4/09
                </div>
                <div
                  class="form-calendar-field--date-description medium"
                >
                  <span
                    class="italic"
                  >
                    Mon
                  </span>
                  <span>
                    , Apr 09, 2018
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="transparent segment"
      >
        <button
          class="button button--fluid larger button--yellow"
          role="submit"
          type="submit"
        >
          AIR_CHANGE__SHOPPING_SEARCH_PAGE__FIND_FLIGHTS_BUTTON
        </button>
      </div>
    </fieldset>
  </form>
</div>
`;

exports[`AirChangeShoppingSearchForm render when isReaccomCoTerminalEligible is true should render correctly with reaccomCoTerminalDepartureAirports and reaccomCoTerminalReturnAirports 1`] = `
<div>
  <form
    class="book-flight-form form widget"
    name="air-change-flight-search"
    novalidate=""
  >
    <fieldset>
      <div
        class="fields grouped"
      >
        <div
          class="field"
        >
          <div
            class="airport-selector"
            data-qa="from"
          >
            <div
              class="clickable-div input"
            >
              <div
                class="input--label"
              >
                <i
                  class="icon icon_airplane-depart"
                />
              </div>
              <div
                class="ellipsis"
              >
                <div
                  class="flex flex-cross-center"
                >
                  <div
                    class="airport-selector--formatted-airport"
                  >
                    AIRPORT NOT FOUND
                  </div>
                  <span
                    class="airport-selector--airport-description"
                  >
                    AIRPORT NOT FOUND, AIRPORT NOT FOUND
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="fields grouped"
      >
        <div
          class="field"
        >
          <div
            class="airport-selector"
            data-qa="to"
          >
            <div
              class="clickable-div input"
            >
              <div
                class="input--label"
              >
                <i
                  class="icon icon_airplane-return"
                />
              </div>
              <div
                class="ellipsis"
              >
                <div
                  class="flex flex-cross-center"
                >
                  <div
                    class="airport-selector--formatted-airport"
                  >
                    AIRPORT NOT FOUND
                  </div>
                  <span
                    class="airport-selector--airport-description"
                  >
                    AIRPORT NOT FOUND, AIRPORT NOT FOUND
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="fields grouped"
      >
        <div
          class="field"
        >
          <div>
            <div
              class="form-calendar-field py3"
              data-qa="depart-and-return-dates"
            >
              <div
                class="form-calendar-field--departure-date"
              >
                <div
                  class="form-calendar-field--date bold xxlarge"
                >
                  4/04
                </div>
                <div
                  class="form-calendar-field--date-description medium"
                >
                  <span
                    class="italic"
                  >
                    Wed
                  </span>
                  <span>
                    , Apr 04, 2018
                  </span>
                </div>
              </div>
              <i
                class="form-calendar-field--icon icon icon_calender"
              />
              <div
                class="form-calendar-field--return-date"
              >
                <div
                  class="form-calendar-field--date bold xxlarge"
                >
                  4/09
                </div>
                <div
                  class="form-calendar-field--date-description medium"
                >
                  <span
                    class="italic"
                  >
                    Mon
                  </span>
                  <span>
                    , Apr 09, 2018
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="transparent segment"
      >
        <button
          class="button button--fluid larger button--yellow"
          role="submit"
          type="submit"
        >
          AIR_CHANGE__SHOPPING_SEARCH_PAGE__FIND_FLIGHTS_BUTTON
        </button>
      </div>
    </fieldset>
  </form>
</div>
`;
